feat: add C ndarray API and refactor `blas/ext/base/dsortins` (#5028)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/base/dsortins/README.md

-   **N**: number of indexed elements.
-   **order**: sort order. If `order < 0.0`, the input strided array is sorted in **decreasing** order. If `order > 0.0`, the input strided array is sorted in **increasing** order. If `order == 0.0`, the input strided array is left unchanged.
-   **x**: input [`Float64Array`][@stdlib/array/float64].
-   **strideX**: stride length.

> The `N` and stride parameters determine which elements in `x` are accessed at runtime. For example, to sort every other element

Reviewer: ```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to sort every other element:
```

* * *

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/blas/ext/base/dsortins.h"
```

#### stdlib_strided_dsortins( N, order, \*X, strideX )

Sorts a double-precision floating-point strided array using insertion sort.

```c
double x[] = { 1.0, -2.0, 3.0, -4.0 };

> stdlib_strided_dsortins( 2, -1, x, 1 );

Reviewer 2: ```suggestion
stdlib_strided_dsortins( 2, -1.0, x, 1 );
```

The input type is a `double`. In which case, include the decimal.

```

The function accepts the following arguments:

-   **N**: `[in] CBLAS_INT` number of indexed elements.
-   **order**: `[in] double` sort order. If `order < 0.0`, the input strided array `x` is sorted in **decreasing** order. If `order > 0.0`, the input strided array `x` is sorted in **increasing** order. If `order == 0.0`, the input strided arrays are left unchanged.
-   **X**: `[inout] double*` input array.
-   **strideX**: `[in] CBLAS_INT` stride length for `X`.

```c

> stdlib_strided_dsortins( const CBLAS_INT N, const double order, double *X, CBLAS_INT strideX );

Reviewer: ```suggestion
stdlib_strided_dsortins( const CBLAS_INT N, const double order, double *X, const CBLAS_INT strideX );
```

```

<!--lint disable maximum-heading-length-->

#### stdlib_strided_dsortins_ndarray( N, order, \*X, strideX, offsetX )

<!--lint enable maximum-heading-length-->

Sorts a double-precision floating-point strided array using insertion sort and alternative indexing semantics.

```c
double x[] = { 1.0, -2.0, 3.0, -4.0 };

> stdlib_strided_dsortins_ndarray( 4, 1, x, 1, 0 );

Reviewer 2: ```suggestion
stdlib_strided_dsortins_ndarray( 4, 1.0, x, 1, 0 );
```

```

The function accepts the following arguments:

-   **N**: `[in] CBLAS_INT` number of indexed elements.

> -   **order**: `[in] CBLAS_INT` sort order.

Reviewer 2: ```suggestion
-   **order**: `[in] double` sort order. If `order < 0.0`, the input strided array `x` is sorted in **decreasing** order. If `order > 0.0`, the input strided array `x` is sorted in **increasing** order. If `order == 0.0`, the input strided arrays are left unchanged.
```

> -   **X**: `[inout] double*` input array. If `order < 0.0`, the input strided array `x` is sorted in **decreasing** order. If `order > 0.0`, the input strided array `x` is sorted in **increasing** order. If `order == 0.0`, the input strided arrays are left unchanged.

Reviewer 2: ```suggestion
-   **X**: `[inout] double*` input array.
```

-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
-   **offsetX**: `[in] CBLAS_INT` starting index for `X`.

```c

> stdlib_strided_dsortins_ndarray( const CBLAS_INT N, const double order, double *X, CBLAS_INT strideX, CBLAS_INT offsetX );

Reviewer: ```suggestion
stdlib_strided_dsortins_ndarray( const CBLAS_INT N, const double order, double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
```

-   **order**: `[in] double` sort order. If `order < 0.0`, the input strided array `x` is sorted in **decreasing** order. If `order > 0.0`, the input strided array `x` is sorted in **increasing** order. If `order == 0.0`, the input strided arrays are left unchanged.
-   **X**: `[inout] double*` input array.
-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
-   **offsetX**: `[in] CBLAS_INT` starting index for `X`.

```c
stdlib_strided_dsortins_ndarray( const CBLAS_INT N, const double order, double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/blas/ext/base/dsortins.h"
#include <stdio.h>

int main( void ) {
    // Create a strided array:
    double x[] = { 1.0, -2.0, 3.0, -4.0, 5.0, -6.0, 7.0, -8.0 };

    // Specify the number of elements:
    int N = 8;

    // Specify a stride:
    int strideX = 1;

    // Sort the array:

>     API_SUFFIX(stdlib_strided_dsortins)( N, 1.0, x, strideX );

Reviewer: ```suggestion
    stdlib_strided_dsortins( N, 1.0, x, strideX );
```
